### PR TITLE
feat: deduplicate drives in drivestore and sdk

### PR DESCRIPTION
**Pull request by Nuhvi**
Stops memory leak of opened Hypercore sessions in Bitkit after: https://github.com/synonymdev/bitkit/pull/963

### CODE REVIEW

File: packages/drive/index.js [hunk 1]
```diff
@@ -61,11 +64,18 @@ export class Drivestore {
    * Get a Hyperdrive by its name.
    */
   get (name = 'public') {
+    const opened = this._openDrives.get(name)
+    if (opened && !opened.core.closed) return opened
```
Comment (dzdidi) [inline packages/drive/index.js:68]: Please, first validate input and then use it  (move below `validateName(name)` call

File: packages/sdk/test/drive.js [hunk 2]
```diff
@@ -7,6 +7,25 @@ import Hyperswarm from 'hyperswarm'
...
+
+  const a = sdk.drive(key)
+  const b = sdk.drive(key)
+
+  t.is(a, b)
+
+  a.core.closed = true
+
+  const c = sdk.drive(key)
+  t.not(c, a)
```
Comment (dzdidi) [inline packages/sdk/test/drive.js:24]: so here `c` is the new drive which will overwrite internal state (`this._openedDrives`)? why not reopen existing closed drive instead?

**Review by dzdidi — CHANGES_REQUESTED** [review]
I suggest to change it to track all drives instead of just open, have methods to close and reopen created drives, and to remove drives from internal state

File: packages/drive/index.js (hunk 1 above)
Comment (Nuhvi) [reply packages/drive/index.js:68]: There isn't really a problem if the name was invalid here, but good practice.

**Comment by Nuhvi** [comment]
Maybe we don't need this PR https://github.com/synonymdev/bitkit/pull/967

**Pull request closed without merging**